Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs

```diff
@@ -163,8 +196,23 @@ private KnobDirection DetectDirection()
 
         private float CalculateValue()
         {
+            if (!subDirectionFound)
+            {
+                float angleX = Mathf.Abs(transform.localRotation.eulerAngles.x - initialLocalRotation.x) % 90;
+                float angleY = Mathf.Abs(transform.localRotation.eulerAngles.y - initialLocalRotation.y) % 90;
+                float angleZ = Mathf.Abs(transform.localRotation.eulerAngles.z - initialLocalRotation.z) % 90;
+                angleX = (Mathf.RoundToInt(angleX) > 89 && Mathf.RoundToInt(angleX) < 91) ? 0 : angleX;
+                angleY = (Mathf.RoundToInt(angleY) > 89 && Mathf.RoundToInt(angleY) < 91) ? 0 : angleY;
+                angleZ = (Mathf.RoundToInt(angleZ) > 89 && Mathf.RoundToInt(angleZ) < 91) ? 0 : angleZ;
+                if (Mathf.RoundToInt(angleX) != 0 || Mathf.RoundToInt(angleY) != 0 || Mathf.RoundToInt(angleZ) != 0)
+                {
+                    subDirection = angleX < angleY ? (angleY < angleZ ? KnobDirection.z : KnobDirection.y) : (angleX < angleZ ? KnobDirection.z : KnobDirection.x);
+                    subDirectionFound = true;
+                }
+            }
+
             float angle = 0;
-            switch (finalDirection)
+            switch (subDirection)
             {
                 case KnobDirection.x:
                     angle = transform.localRotation.eulerAngles.x - initialLocalRotation.x;
```

Comment:
A few lines below (GitHub won't let me comment outside the patch) you round * 1000 then divide back by 1000. Math.Round has a two argument form where you can provide a precision. Math.Round(angle, 3), for example.